xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/flamingo.js

```diff
@@ -40,14 +40,78 @@ require(['jquery', 'bootstrap'], function($) {
         }
       }
     });
-
-    // When the drawer is close, collapse sub items
-    $(body).on('drawer.closed', function() {
-      $('.drawer-menu-sub-item').removeClass('in').attr('aria-expanded', 'false');
-    });
-
     // Activate the popover when hovering the Translate button.
     var translateButton = $('#tmTranslate [data-toggle="popover"]');
     translateButton.attr('title', translateButton.attr('data-title')).popover();
   });
 });
+
+/*
+  Handle the behavior of all drawers on the page.
+  In order to create a drawer that is compatible with this script, two elements are necessary:
+  * A drawer toggler button, that will have:
+   ** a reference to the drawer container in its attribute 'aria-control'
+   ** the class 'drawer-toggle'
+  * A drawer container, that will have:
+   ** a unique ID
+   ** the class 'drawer-nav' (for style) 
+   ** an accessible name 
+   ** The content to display inside.
+  In addition, the drawer can contain a `.drawer-close` button that will close the drawer on click.
+  For an example of drawer creation, see #tmDrawerActivator and #tmDrawer.
+ */
+require(['jquery'], function($) {
+  $('.drawer-toggle').each(function(index) {
+    // Setting up the drawer.
+    let drawerContainerToggler = $(this);
+    let drawerId = drawerContainerToggler.attr('aria-controls');
+    let drawerContainer = $(document.getElementById(drawerId));
+    
+    let openDrawer = () => drawerContainer.trigger('drawer' + index + '.opened');
+    let closeDrawer = () => drawerContainer.trigger('drawer' + index + '.closed');
+    drawerContainerToggler.on('click', openDrawer);
+    // Close drawer when clicking the backdrop (or any element outside of the drawer itself).
+    drawerContainer.on('click', (event) => {
+      let drawerzone = event.target.getBoundingClientRect();
+      if (drawerzone.left > event.clientX || drawerzone.right < event.clientX
+        || drawerzone.top > event.clientY || drawerzone.bottom < event.clientY) {
```

Comment:
The [Stack Overflow discussion](https://stackoverflow.com/questions/25864259/how-to-close-the-new-html-dialog-tag-by-clicking-on-its-backdrop) around this solution mentions several problems, in particular that click events from form submissions might bubble up and might not have correct coordinates when they originate from pressing "enter". It is mentioned that as a workaround, it should be checked if the event target is actually the dialog. Is there a reason why you didn't add that check?

Reply:
So far there's only links in the drawer. But if we want to generalize their use we need to be ready for it to contain a form.

I addressed this issue in https://github.com/xwiki/xwiki-platform/pull/2282/commits/b117cb9d7baa632bfcfd396ec6a13385b2ba38cd :+1: